docker_images/k2_sherpa/app/common.py

```diff
@@ -0,0 +1,201 @@
+import functools
+import k2
+import kaldifeat
+import torch
+import json
+import sentencepiece as spm
+
+from huggingface_hub import HfApi, hf_hub_download
+from sherpa import RnntConformerModel
+from typing import List, Optional, Union
+
+from .decode import (
+    run_model_and_do_greedy_search,
+    run_model_and_do_modified_beam_search,
+)
+
+
+def get_hfconfig(model_id, config_name="hf_demo"):
+    info = HfApi().model_info(repo_id=model_id)
+    config_file = hf_hub_download(model_id, filename="config.json")
+    with open(config_file) as config:
+        info.config = json.load(config)
+
+    if info.config and config_name is not None:
+        if config_name in info.config:
+            return info.config[config_name]
+        else:
+            raise ValueError("Config section " + config_name + " not found")
+    else:
+        return info
+
+
+def model_from_hfconfig(hf_repo, hf_config):
+    nn_model_filename = hf_hub_download(hf_repo, hf_config["nn_model_filename"])
+    token_filename = (
+        hf_hub_download(hf_repo, hf_config["token_filename"])
+        if "token_filename" in hf_config
+        else None
+    )
+    bpe_model_filename = (
+        hf_hub_download(hf_repo, hf_config["bpe_model_filename"])
+        if "bpe_model_filename" in hf_config
+        else None
+    )
```

Comment:
I think you need to check that `token_filename` and `bpe_model_filename` cannot be `None` at the same time.